### ISSUE TITLE
[Feature Request] Option for whether auto reload file when file content on disk changes

### ISSUE THREAD

**Issue by WilliamStone** [Microsoft/vscode#41250]
- VSCode Version: Code - Insiders 1.20.0-insider (c4065a8ba5261456338f850a37e67732d98bf57a, 2018-01-05T08:06:27.790Z)
- OS Version: Windows_NT x64 10.0.16299
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
markdown-extension-pack|bat|0.1.2
vscode-markdownlint|Dav|0.12.0
todopomo|Eas|0.2.5
vscode-projects-plus|fab|1.15.0
vscode-projects-plus-todo-plus|fab|1.4.0
vscode-todo-plus|fab|1.7.2
code-runner|for|0.8.7
mdmath|goe|2.1.0
rest-client|hum|0.16.2
plantuml|jeb|2.4.0
HOCON|sab|0.0.1
todotasks|san|0.5.0
markdown-preview-enhanced|shd|0.3.1
vscode-todo-highlight|way|0.5.11
markdown-pdf|yza|0.1.7
markdown-all-in-one|yzh|1.0.0

---

Steps to Reproduce:

1. Open a file in VSCode editor
2. Use another editor tool to modify the file content
3. VSCode editor automatically reloads file content without asking

This auto-reload behavior happens when there's no unsaved content in VSCode editor. In most times it is covenient, however in some cases it is unwanted to silently auto-reload file modified by another program. 

I'll describe one such case: When I'm editing a file(XXX.txt) located in cloud storage e.g. dropbox, the file will be uploaded every time I save it. If I do 2nd save before file upload triggered by 1st save finish, dropbox will detect this situation as a conflict and it will rename current file with 2nd save content to something like XXX-conflict-<timestamp>-<host>.txt, and restore content of XXX.txt to that of 1st save. And then VSCode editor reloads XXX.txt without asking for permission, effectively restored editor content to that of 1st save.

So I kindly request to give such auto-reload feature an option to control, with 3 options: 
* silently reload
* never reload(but somehow prompt user file content on disk changed)
* ask user what to do. 

Basically it can be a global option that affects all files and editors; Or it could be better to be option per file, per directory, or per path pattern for fine-grained control over auto-reload behavior.
Configurable path pattern seems rationale, For example, in the case described above, all file in dropbox should never reload, while I prefer other files silently auto reloaded to save manual operation.

Nevertheless, thanks for your great job to bring the world such powerful and wonderful editor!

Reproduces without extensions: Yes

**Comment by WilliamStone** [comment]
Rechecked issue list and found the following 2 similar issues:

#14298
#20876

However IMO they are not fully duplicate of this issue, since in this issue I distinguished a scenario that  needs control not by file type, but by file path pattern (only files under certain folder shall disable auto-reload). So I wish this feature not get ignored by marking this issue duplicate.

**Comment by bpasero** [comment]
I still think its a duplicate though but please comment on one of the other issues and we can update them.

**Comment by WilliamStone** [comment]
I found another issue that i feel more close to this one: #23107, since it is the only one that discusses option for controlling auto-reload behavior, so I commented under it.
